src/server.c

```diff
@@ -400,19 +400,50 @@ int dictExpandAllowed(size_t moreMem, double usedRatio) {
     }
 }
 
-/* Adds dictionary to the rehashing list in cluster mode, which allows us
+/* Updates the bucket count in cluster-mode for the given dictionary in a DB. bucket count
+ * incremented with the new ht size during the rehashing phase.
+ * And also adds dictionary to the rehashing list in cluster mode, which allows us
  * to quickly find rehash targets during incremental rehashing.
- * In non-cluster mode, we don't need this list as there is only one dictionary per DB. */
+ * 
+ * In non-cluster mode, bucket count can be retrieved directly from single dict bucket and
+ * we don't need this list as there is only one dictionary per DB. */
 void dictRehashingStarted(dict *d) {
     if (!server.cluster_enabled || !server.activerehashing) return;
+
+    unsigned long long from, to;
+    dictRehashingInfo(d, &from, &to);
+    server.db[0].sub_dict[DB_MAIN].bucket_count += to; /* Started rehashing (Add the new ht size) */
+    if (from == 0) return; /* No entries are to be moved. */
     listAddNodeTail(server.db[0].sub_dict[DB_MAIN].rehashing, d);
 }
 
+/* Updates the bucket count for the given dictionary in a DB. It removes
+ * the old ht size of the dictionary from the total sum of buckets for a DB.  */
+void dictRehashingCompleted(dict *d) {
+    if (!server.cluster_enabled || !server.activerehashing) return;
```

Comment:
Is the `activerehashing` check needed here? I think it's unwanted

Reply:
I wonder about the same check in the other method.
I'm AFK so can't validate, maybe you can explain it to me.
When there's a fork active, we don't wanna start, or even perform incremental rehashing, but that method isn't the one preventing it, so that early return can mean that the buckets will be allocated but we won't count them.
What am I missing?

Reply:
I initially started incrementing the bucket_count before that check however the only challenge with the `rehashingStarted` cb  is that the dict gets added to the `rehashing` list only if rehashing is enabled. So, avoided double counting.

I think adding the dict to the rehashing list was avoid earlier to avoid the list to grow while no action is being taken.

Reply:
Now that I think more about this, dict rehashing could get completed via single bucket rehashing during read/write operation and we would miss updating the bucket count. Let me think more about it.

Reply:
Handled this. Thanks @oranagra 🙏